chain/types/src/lib.rs

```diff
@@ -230,6 +230,100 @@ impl<M: Middleware> ContractInstances<M> {
             module_implementation,
         })
     }
+
+    /// Deploys debugging environment for testing purpose (with safe network registry proxy) via the given provider.
+    pub async fn deploy_for_testing_debug(provider: Arc<M>, deployer: &ChainKeypair) -> Result<Self, ContractError<M>> {
```

Comment:
Also, if this is not intended for production use, please, hide it behind a `testing` feature.

Reply:
I realized that there's a `cfg!(test)` for deploying dummy proxy. So this function cannot even be used in testing

Reply:
Removed this unused funciton altogether